### PR TITLE
Les 97 Create modal for enter mobile phone

### PR DESCRIPTION
**Pull request by OxygenO22**

### CODE REVIEW

**Review by Evgeniy121212 — COMMENTED** [review]
please, remove useless files from PR:

image

**Review by Kurilo-Pavel — COMMENTED** [review]
const [phone, Phone] = useState<string>('');
  const [phoneDirty, PhoneDirty] = useState<boolean>(false);
  const [phoneError, PhoneError] = useState<string>('The input field must be filled');

add 'set' before Phone
example [phone, setPhone]=useState('')

**Review by Kurilo-Pavel — COMMENTED** [review]
if useState is boolean add 'is' example 'isPhoneDirty'

File: src/components/ModalEnterPhone.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  const sendphone = () => {
+    setActive(false);
+  }
+
+  const blurHandler = () => {
+    PhoneDirty(true);
+  }
+
+  const phoneHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const re = /^(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/;
```
Comment (Kurilo-Pavel) [inline src/components/ModalEnterPhone.tsx:38]: i think you can use "React-Phone-Input-2'

File: src/components/ModalEnterPhone.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import { connected } from 'process';
+import React,  {FC, useEffect, useState} from 'react';
+import './ModalEnterPhone.scss';
+
```
Comment (Kurilo-Pavel) [inline src/components/ModalEnterPhone.tsx:4]: You should use completed component PasswordAndConfirm

File: src/components/ModalEnterPhone.tsx (hunk 1 above)
Comment (OxygenO22) [reply src/components/ModalEnterPhone.tsx:38]: But it work

File: src/components/ModalEnterPhone.tsx (hunk 2 above)
Comment (OxygenO22) [reply src/components/ModalEnterPhone.tsx:4]: What it mean?

**Comment by OxygenO22** [comment]
> const [phone, Phone] = useState(''); const [phoneDirty, PhoneDirty] = useState(false); const [phoneError, PhoneError] = useState('The input field must be filled');
> 
> add 'set' before Phone example [phone, setPhone]=useState('')



> please, remove useless files from PR:
> 
> image

I think, I fixed all comments, please appruwe

File: src/components/ModalEnterPhone.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
+* {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    font-size: 16px;
+}
+
+$main-color: #2467a3;
```
Comment (Terapool) [inline src/components/ModalEnterPhone.scss:8]: Let's call "$primary-color"

File: src/components/ModalEnterPhone.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    setActive,
+  }) => {
+  
+  const [phone, setPhone] = useState<string>('');
+  const [isPhoneDirty, setPhoneDirty] = useState<boolean>(false);
+  const [phoneError, setPhoneError] = useState<string>('The input field must be filled');
+  const [formValid, setFormValid] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (phoneError) {
```
Comment (Terapool) [inline src/components/ModalEnterPhone.tsx:21]: setFormValid(!phoneError) without if

File: src/components/ModalEnterPhone.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  setActive: (bool: boolean) => void ;
+}
+
+const ModalEnterPhone: FC<CardProps> = 
+  ({
+    active, 
+    setActive,
+  }) => {
+  
+  const [phone, setPhone] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/components/ModalEnterPhone.tsx:15]: вщту

File: src/components/ModalEnterPhone.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
+import React,  {FC, useEffect, useState} from 'react';
+import './ModalEnterPhone.scss';
+
+interface CardProps {
+  active: boolean;
+  setActive: (bool: boolean) => void ;
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/components/ModalEnterPhone.tsx:6]: done

File: src/components/ModalEnterPhone.scss (hunk 3 above)
Comment (OxygenO22) [reply src/components/ModalEnterPhone.scss:8]: done

File: src/components/ModalEnterPhone.tsx (hunk 4 above)
Comment (OxygenO22) [reply src/components/ModalEnterPhone.tsx:21]: done

**Review by Terapool — APPROVED** [review]

**Review by Kurilo-Pavel — APPROVED** [review]

**Review by valer-sky — APPROVED** [review]

**Comment by Evgeniy121212** [comment]
@OxygenO22, 
image

**Pull request merged**